Dev (#1)
AllenInstitute/BCDC-Dashboards: README.md

### Workflow
1. Input: BCDC-Metadata
2. Data Wrangling: dashboard_data_wrangling.ipynb
    Output: dash_count_df.csv
3. Dashboard script: tast_dash_counts.py (takes dash_count_df.csv as input)
    Output: Dashboard page with visualizations

### Business logics:

[dashboard_logics.pdf](https://github.com/AllenInstitute/BCDC-Dashboards/files/11033385/dashboard_logics.pdf)

Currently, we are aiming to count the below units:
1. Number of samples received for each sample type per group - for example, how many brain/cell related samples did we receive per quarter; or for each grant?
2. Number of donors/subjects per group - for example, how many donors/subjects per quarter/modality/technique?
3. Number of subspecimens (brains, cells, tissue samples, etc.) processed per quarter/grant/technique/etc.

> While counting, the groups and categories will be collapsed into similar terminologies - for example, 'whole cell' and 'cell body' both mean the same thing, and as a result, they will be collapsed together under 'cells' and their counts will be counted towards the same category. 

Reviewer: I'm not sure that `whole cell` and `cell body` mean the same thing. IIRC, `cell body` is a soma, `whole cell` would include dendrites and axons (e.g.).

> While counting, the groups and categories will be collapsed into similar terminologies - for example, 'whole cell' and 'cell body' both mean the same thing, and as a result, they will be collapsed together under 'cells' and their counts will be counted towards the same category. 

Reviewer: ```suggestion
For simplicity of counting, we aim to group similar categories together. For example, `whole cell` and `cell body` will be grouped together as they both involve counting of the same general type of entity (a cell, either in part or whole). As a result, their counts will be summed in a single category.
```

Reviewer: > Suggested change
> While counting, the groups and categories will be collapsed into similar terminologies - for example, 'whole cell' and 'cell body' both mean the same thing, and as a result, they will be collapsed together under 'cells' and their counts will be counted towards the same category. 
> For simplicity of counting, we aim to group similar categories together. For example, `whole cell` and `cell body` will be grouped together as they both involve counting of the same general type of entity (a cell, either in part or whole). As a result, their counts will be summed in a single category.

This is just a suggestion, feel free to wordsmith.

1. Logics to collapse subspecimen types for counts:

```
specimen_count_map = {
    'brains' : ['whole brain',
                'brain', 'brains'], # fix the plural
    'cells' : ['whole cell', 
               'cell', 'cells', 'cell body', 'cell suspension', 'cell nucleus', 'nuclei', 'resconstruction', 'cell in slice'],
    'nucleus' : ['cell nucleus', 

>                  'nuclei'], # collapsed nucleus and cells above in 'cells' - however, do we still want a separate 'nucleus' count?

Reviewer: Hmm, tough to say. Nucleus is part of a cell, so given the above logic regarding collapsing parts of cells (i.e., cell body) to cell, nucleus would also be collapsed. OTOH, many techniques target nuclei rather than cell, so I can see the utility.

    'library' : ['library'],
    'tissue samples' : ['brain hemisphere', 
                        'brain region', 
                        'brain section set', 
                        'brain slice', 

>                         'library', 

Reviewer: Is `library` occurring twice in different locations?

                        'spinal cord slice',
                        'tissue sample']   
}
```

> Since specimen types such as 'brain' and 'whole brain' mean the same thing, they are being collapsed into the same categories and their according numbers of processed specimens will be summed and counted towards the same category.

Reviewer: Again, technically `brain` and `whole brain` don't mean the same thing. 
```suggestion
For simplicity, we group specimen types such as 'brain' and 'whole brain' together and the number of processed specimens will be summed and counted towards the same category.
```
This seems more accurate to me.

2. Logics to collapse species for counts:

```
species_map = {

>     'primates' : ['chimpanzee', 'small-eared galago', 'western gorilla', 'green monkey', 'pig-tailed macaque',

Reviewer 2: This category could be "non-human primates" or "nhp" if humans are excluded since humans are technically primates.

                 "ma's night monkey", 'rhesus macaque', 'bolivian squirrel monkey',
       'crab-eating macaque'],
    'humans' : ['human'], ## mostly mice and humans
    'mice' : ['mouse'],

>     'marmoset': ['marmoset'],

Reviewer: Marmosets are primates.

>     'small mammals': [

Reviewer: Many of the primates are also small mammals, so this could be confusing. Maybe `non-primate small mammals` or something?

Author: changed this to 'other mammals' - opening a new PR soon
